html/includes/authentication/ldap.inc.php

```diff
@@ -217,10 +217,42 @@ function can_update_users()
 
 function get_user($user_id)
 {
-    foreach (get_userlist() as $user) {
-        if ($user['user_id'] === $user_id) {
+    try {
+        $connection = get_ldap_connection();
+
+        $filter = '(' . Config::get('auth_ldap_uid_attribute') . '=' . $user_id . ')';
+        $search = ldap_search($connection, trim(Config::get('auth_ldap_suffix'), ','), $filter);
+        $entries = ldap_get_entries($connection, $search);
+
+        if ($entries['count']) {
+            foreach ($entries as $entry) {
+                $username = $entry['uid'][0];
+                $realname = $entry['cn'][0];
+                $uid_attr = strtolower(Config::get('auth_ldap_uid_attribute', 'uidnumber'));
+                $user_id = $entry[$uid_attr][0];
+                $email = $entry[Config::get('auth_ldap_emailattr', 'mail')][0];
+                $ldap_groups = get_group_list();
+                foreach ($ldap_groups as $ldap_group) {
+                    $ldap_comparison = ldap_compare(
+                        $connection,
+                        $ldap_group,
+                        Config::get('auth_ldap_groupmemberattr', 'memberUid'),
+                        get_membername($username)
+                    );
+                    if (!Config::has('auth_ldap_group') || $ldap_comparison === true) {
+                        $userlist[$username] = array(
+                            'username' => $username,
+                            'realname' => $realname,
+                            'user_id' => $user_id,
+                            'email' => $email,
+                        );
+                    }
+                }
+            }
             return $user;
```

Comment:
$user doesn't seem to be set?  Also, the return should probably be inside the foreach loop.

Reply:
Fixed the return in the latest commit.  The LDAP query should only return 1 result, so inside or outside the foreach shouldn't matter.  I'm not sure what we should do if the LDAP query returns more than one result.